lib/upipe-srt/upipe_srt_receiver.c

```diff
@@ -0,0 +1,1454 @@
+    UBASE_HANDLED_RETURN(upipe_srt_receiver_output_control_output(upipe, command, args));
+    switch (command) {
+        case UPIPE_SET_FLOW_DEF: {
+            struct uref *flow_def = va_arg(args, struct uref *);
+            return upipe_srt_receiver_output_set_flow_def(upipe, flow_def);
+        }
+        default:
+            return UBASE_ERR_UNHANDLED;
+    }
+}
+static int upipe_srt_receiver_output_control(struct upipe *upipe, int command, va_list args)
+{
+    UBASE_RETURN(_upipe_srt_receiver_output_control(upipe, command, args))
+    return upipe_srt_receiver_output_check(upipe, NULL);
+}
+
+/** @internal @This initializes the output manager for a srt set pipe.
+ *
+ * @param upipe description structure of the pipe
+ */
+static void upipe_srt_receiver_init_sub_mgr(struct upipe *upipe)
+{
+    struct upipe_srt_receiver *upipe_srt_receiver = upipe_srt_receiver_from_upipe(upipe);
+    struct upipe_mgr *sub_mgr = &upipe_srt_receiver->sub_mgr;
+    sub_mgr->refcount = upipe_srt_receiver_to_urefcount_real(upipe_srt_receiver);
+    sub_mgr->signature = UPIPE_SRT_RECEIVER_OUTPUT_SIGNATURE;
+    sub_mgr->upipe_alloc = upipe_srt_receiver_output_alloc;
+    sub_mgr->upipe_input = NULL;
+    sub_mgr->upipe_control = upipe_srt_receiver_output_control;
+}
+
+
+/** @internal @This allocates a SRT receiver pipe.
+ *
+ * @param mgr common management structure
+ * @param uprobe structure used to raise events
+ * @param signature signature of the pipe allocator
+ * @param args optional arguments
+ * @return pointer to upipe or NULL in case of allocation error
+ */
+static struct upipe *upipe_srt_receiver_alloc(struct upipe_mgr *mgr,
+                                        struct uprobe *uprobe,
+                                        uint32_t signature, va_list args)
+{
+    struct upipe *upipe = upipe_srt_receiver_alloc_void(mgr, uprobe, signature, args);
+    struct upipe_srt_receiver *upipe_srt_receiver = upipe_srt_receiver_from_upipe(upipe);
+
+#ifdef UPIPE_HAVE_GCRYPT_H
+    if (!gcry_control(GCRYCTL_INITIALIZATION_FINISHED_P)) {
+        uprobe_err(uprobe, upipe, "Application did not initialize libgcrypt, see "
+        "https://www.gnupg.org/documentation/manuals/gcrypt/Initializing-the-library.html");
+        upipe_srt_receiver_free_void(upipe);
+        return NULL;
+    }
+#endif
+
+    upipe_srt_receiver_init_urefcount(upipe);
+    upipe_srt_receiver_init_urefcount_real(upipe);
+    upipe_srt_receiver_init_sub_outputs(upipe);
+    upipe_srt_receiver_init_sub_mgr(upipe);
+
+    upipe_srt_receiver_init_uref_mgr(upipe);
+    upipe_srt_receiver_init_ubuf_mgr(upipe);
+    upipe_srt_receiver_init_output(upipe);
+
+    upipe_srt_receiver_init_upump_mgr(upipe);
+    upipe_srt_receiver_init_upump_timer(upipe);
+    upipe_srt_receiver_init_upump_timer_lost(upipe);
+    upipe_srt_receiver_init_uclock(upipe);
+    upipe_srt_receiver_require_uclock(upipe);
+
+    // FIXME
+    upipe_srt_receiver->socket_id = 0;
+    upipe_srt_receiver->control = NULL;
```

Comment:
Clarify what the FIXME is about.  I think I saw a comment in handshake that the socket id would be set after receiving the first packet.  If the same applies here maybe you should copy that comment.